specs/progress.md

```diff
@@ -0,0 +1,142 @@
+# fast-progress
+
+## Overview
+
+*Progress* is used to display the length of a process or to represent an unspecified wait time. *Progress* can either be circular or linear. *Progress* can either be in an **indeterminate** or **determinate** state. This is determined by whether `value` attribute is passed to the component. 
+
+### Use Cases
+
+The *progress* component should be used for an element that displays the progress status for a task that takes a long time or consists of several steps. Typical use cases include loading elements or status of a download. 
+  
+### Features
+
+A progress should allow the following properties:
+- `value`, the value of the progress, if not passed the progress will be put into its "indeterminate" state.
+- `min-value`, the minimum value.
+- `max-value`, the maximum value.
+- `circular`, whether the progress renders circular or not.
+- `paused`, whether the progress is paused or not.
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/progress)
+- [Material UI](https://material-ui.com/components/progress/)
+- [Lightning Design / Linear](https://www.lightningdesignsystem.com/components/progress-bar/)
+- [Lightning Design / Circular](https://www.lightningdesignsystem.com/components/progress-ring/)
+- [Ant Design](https://ant.design/components/progress/)
+- [Atlassian / Linear](https://atlaskit.atlassian.com/packages/server/progress-bar)
+- [Atlassian / Circular](https://atlaskit.atlassian.com/packages/core/spinner)
+- [Windows (UWP)](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/progress-controls)
+
+---
+
+### API
+
+*Component name:*
+- `fast-progress`
+
+*Attributes:*
+- `value`: number
+- `min-value`: number
+- `max-value`: number
+- `circular`: boolean
+- `paused`: boolean
+
+### Anatomy and Appearance
+
+*Parts:*
+- determinate
+- indeterminate
+- circular-container
+- circular-background
+- circular-determinate
+- circular-indeterminate
+- linear-determinate
+- linear-indeterminate-indicator-1
+- linear-indeterminate-indicator-2
+
+*Circular Determinate Template:*
+```
+<div
+    role="progressbar"
+    aria-valuenow={50}
+    aria-valuemin={0}
+    aria-valuemax={100}
+>
+    <div part="determinate" slot={"determinate"}>
+        <svg part="circular-container" viewBox="0 0 16 16">
+            <circle part="circular-background" cx="8px" cy="8px" r="7px" />
+            <circle part="circular-determinate" style={{stroke-dasharray: "22px 44px"}} cx="8px" cy="8px" r="7px" />
+        </svg>
```

Comment:
We seem to be boxing this style in via these SVGs, @bheston can you give an opinion on how we should want to allow users to style this component?

Reply:
An important note about the aria attributes (and I can't remember if I saw this in other specs or not): we'll want the aria attributes to exist on the custom element host itself, rather than the nodes inside the shadow dom.

Reply:
Had an in depth conversation with @EisenbergEffect on this topic so I thought I would summarize here:

We will do work in our components to write aria values to the host element based on attributes that come from the app author. So an app author would implement as you would expect:

```
<fast-progress
    value={50}
    min-value={0}
    max-value={100}
/>
```

But the DOM will show the component as:

```
<fast-progress
    value={50}
    min-value={0}
    max-value={100}
    aria-valuenow={50}
    aria-valuemin={0}
    aria-valuemax={100}
/>
```

and the snapshot of the html inside the shadow root as:

```
<div
    role="progressbar"
    aria-valuenow={50}
    aria-valuemin={0}
    aria-valuemax={100}
>
    <div part="determinate" slot={"determinate"}>
        <div part="linear-determinate" style={{ width: 50% }}/>
    </div>
</div>
```

This spec is correct as written because I'm showing "snap shots" of the DOM under **Anatomy and Appearance** and showing author implementation under the **Implementation** section.

Reply:
Cool, the aria attributes sounds like what I suggested above.

I do think it's reasonable that people may want to restyle the svg, but it sounds like that will be possible with a custom template. This would be for someone who wants to do something more graphical or formatted slightly differently. They would still want to respect the value attribute (probably), but that would be up to them.